examples/apps/cli/main.c

```diff
@@ -98,7 +98,7 @@ int main(int argc, char *argv[])
     {
         alarm(0);
 #if OPENTHREAD_ENABLE_COVERAGE
-        __gcov_flush();
```

Comment:
We need to call flash or dump according to the current compiler version?

Reply:
Yes, I use this PR to check if the __gcov_dump is available in all versions.